src/audio/igo_nr/igo_nr.c

```diff
@@ -275,124 +277,97 @@ static inline int32_t set_capture_func(struct comp_dev *dev)
+			 bs, SOF_IGO_NR_MAX_SIZE);
+		return -EINVAL;
 	}
 
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev)
-		return NULL;
-
-	dev->ipc_config = *config;
-
 	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
 	if (!cd)
-		goto fail;
+		return -ENOMEM;
 
+	md->private = cd;
 	ret = IgoLibGetInfo(&cd->igo_lib_info);
 	if (ret != IGO_RET_OK) {
-		comp_cl_err(&comp_igo_nr, "igo_nr_new(): IgoLibGetInfo() Failed.");
+		comp_err(dev, "igo_nr_init(): IgoLibGetInfo() Failed.");
+		ret = -EINVAL;
 		goto cd_fail;
 	}
 
 	cd->p_handle = rballoc(0, SOF_MEM_CAPS_RAM, cd->igo_lib_info.handle_size);
 	if (!cd->p_handle) {
-		comp_cl_err(&comp_igo_nr, "igo_nr_new(): igo_handle memory rballoc error");
+		comp_err(dev, "igo_nr_init(): igo_handle memory rballoc error for size %d",
+			 cd->igo_lib_info.handle_size);
+		ret = -ENOMEM;
 		goto cd_fail;
 	}
 
-	comp_set_drvdata(dev, cd);
-
 	/* Handler for configuration data */
 	cd->model_handler = comp_data_blob_handler_new(dev);
 	if (!cd->model_handler) {
-		comp_cl_err(&comp_igo_nr, "igo_nr_new(): comp_data_blob_handler_new() failed.");
+		comp_err(dev, "igo_nr_init(): comp_data_blob_handler_new() failed.");
 		goto cd_fail;
 	}
 
 	/* Get configuration data */
-	ret = comp_init_data_blob(cd->model_handler, bs, ipc_igo_nr->data);
+	ret = comp_init_data_blob(cd->model_handler, bs, cfg->data);
 	if (ret < 0) {
-		comp_cl_err(&comp_igo_nr, "igo_nr_new(): comp_init_data_blob() failed.");
+		comp_err(dev, "igo_nr_init(): comp_init_data_blob() failed.");
 		goto cd_fail;
 	}
-	dev->state = COMP_STATE_READY;
 
-	comp_cl_info(&comp_igo_nr, "igo_nr created");
+	/* update downstream (playback) or upstream (capture) buffer parameters */
+	mod->verify_params_flags = BUFF_PARAMS_RATE;
+
+	comp_info(dev, "igo_nr created");
 
-	return dev;
+	return 0;
 
 cd_fail:
 	comp_data_blob_handler_free(cd->model_handler);
 	rfree(cd);
-fail:
-	rfree(dev);
-	return NULL;
+
+	return ret;
 }
 
-static void igo_nr_free(struct comp_dev *dev)
+static int igo_nr_free(struct processing_module *mod)
 {
-	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_data *cd = module_get_private_data(mod);
 
-	comp_info(dev, "igo_nr_free()");
+	comp_info(mod->dev, "igo_nr_free()");
 
 	comp_data_blob_handler_free(cd->model_handler);
 
 	rfree(cd->p_handle);
 	rfree(cd);
-	rfree(dev);
-}
-
-static int32_t igo_nr_verify_params(struct comp_dev *dev,
-				    struct sof_ipc_stream_params *params)
-{
-	int32_t ret;
-
-	comp_dbg(dev, "igo_nr_verify_params()");
-
-	/* update downstream (playback) or upstream (capture) buffer parameters
-	 */
-	ret = comp_verify_params(dev, BUFF_PARAMS_RATE, params);
-	if (ret < 0)
-		comp_err(dev, "igo_nr_verify_params(): comp_verify_params() failed.");
-
-	return ret;
+	return 0;
 }
 
-/* set component audio stream parameters */
-static int32_t igo_nr_params(struct comp_dev *dev,
-			     struct sof_ipc_stream_params *pcm_params)
+#if DISABLED_CODE
```

Comment:
is `DISABLED_CODE` defined anywhere? Checking with an `#if` means that the macro is defined and you're checking for it to be equal to 0, I've seen builds failing when an undefined macro is used like this, possibly it depends on the compiler version or on compilation flags. Just use `#ifdef` unless you're sure that it's defined

Reply:
We can change this define to something like ```IGO_NR_IPC4_INTEGRATION_TODO``` as place a large comment next to the macro declaration stating this is work that still needs to be done for ipc4/module API that needs to be done by someone with the latest IGO NR binaries (and know what they are doing).

Reply:
I think I can fix this for audio stream type module adapter client even if it's temporary version.